packages/genesys-spark-components/src/components/stable/gux-popover/gux-popover.tsx

```diff
@@ -118,10 +117,7 @@ export class GuxPopover {
     // This is 13 because this makes the arrow look aligned
     const arrowLen = 13;
 
-    if (
-      this.popupElement &&
-      getContainingDocument(this.root).getElementById(this.for)
-    ) {
+    if (this.popupElement && findElementById(this.root, this.for)) {
```

Comment:
I recommend doing this operation once and checking if it was successful instead of repeating it in the for loop. Also, it looks like more or less the same code is in `gux-popover-list`.

Reply:
I dont think executing this just once would work as there would be cases where its dynamically removed?

Reply:
What I mean is, put it in a variable immediately before the if, then check the variable. Only synchronous logic so there shouldn't be any problem with it being removed.

```
private updatePosition(): void {
  const forElement = findElementById(this.root, this.for);
  // This is 13 because this makes the arrow look aligned
  const arrowLen = 13;

  if (this.popupElement && forElement) {
```